### ISSUE TITLE
Override showing of placeholder on Multiple Input selection

### ISSUE THREAD

**Issue by happysomani** [ericgio/react-bootstrap-typeahead#684]
**Is your feature request related to a problem? Please describe.**
I would like to show a placeholder in the input even if a user has selected a few items previously.  Currently in the code [here](https://github.com/ericgio/react-bootstrap-typeahead/blob/master/src/components/TypeaheadInputMulti.tsx#L52) the code clears out the placeholder text if the length of the selected items is greater than zero.  

But for e.g. The default placeholder text could be "Search for states..." when there is no state selected.  Once a user has some state selected in the typeahead box, I may want to update the placeholder text to something like "Search for additional states"

**Describe the solution you'd like**
Provide an override prop that would always show the placeholder regardless whether the number of selected items is zero or not.

**How is this solution useful to others?**
Others may be able to also use the placeholder to provide more context to the users using the typeahead.

**Describe alternatives you've considered**
None.

**Additional context**
The feature would show up something like this 

![image](https://user-images.githubusercontent.com/7295715/149038685-73ab7d90-7f25-4563-9119-67747fdb79b4.png)

**Comment by happysomani** [comment]
@ericgio can you please help review the above feature request 🙏  ?

**Comment by ericgio** [comment]
Hi @happysomani, thank you for the feature request. It's a reasonable one; I think the best approach would be to move the placeholder logic up to the main Typeahead component. That would allow developers like yourself to render the multi-select input using `renderInput` and have control over the placeholder value.

**Comment by happysomani** [comment]
I may not have understood your recommendation around the implementation approach.  Can you please explain in slightly more detail please?

**Comment by ericgio** [comment]
@happysomani: Sorry for not being clear, my comment was mostly a note to myself. I'm suggesting moving the line you highlighted up to [here](https://github.com/ericgio/react-bootstrap-typeahead/blob/master/src/components/Typeahead.tsx#L264). That would keep the current behavior by default, but allow you to use the `TypeaheadInputMulti` component along with `renderInput` to customize the behavior.

**Comment by happysomani** [comment]
I can make the change and submit a pull request if you would prefer that.

**Comment by harrylepotter-win** [comment]
@happysomani also interested in this functionality

**Comment by ColinGoldie** [comment]
This would greatly enhance usability. +1 from me!!!!

**Comment by ericgio** [comment]
Fixed in v6.0.0-rc.1. You can override the default behavior using `renderInput`:

```jsx
<Typeahead
  ...
  renderInput={(inputProps) => (
    <TypeaheadInputMulti {...inputProps} />
  )}
/>
```
